pkg/scheduler/invoker/remoteinvoker.go

```diff
@@ -0,0 +1,186 @@
+	var errorReason string
+
+	err = i.marshalHTTPResponse(body, respModel, params)
+	if err == nil {
+		errorReason = respModel.Error
+	} else {
+		i.reportUnmarshalError(resp.StatusCode, body, err)
+		errorReason = fmt.Sprintf("received unsupported reconciler response (HTTP code: %d): %s",
+			resp.StatusCode, string(body))
+	}
+
+	return i.updateOperationState(params, model.OperationStateClientError, errorReason)
+}
+
+func (i *RemoteReconcilerInvoker) reportUnmarshalError(httpCode int, body []byte, err error) {
+	i.logger.Warnf("Remote invoker: Failed to unmarshal reconciler response (HTTP-code: %d / Body: %s): %s",
+		httpCode, string(body), err)
+}
+
+func (i *RemoteReconcilerInvoker) sendHTTPRequest(params *Params) (*http.Response, error) {
+	component := params.ComponentToReconcile.Component
+
+	callbackURL := fmt.Sprintf(callbackURLTemplate,
+		i.config.Scheme,
+		i.config.Host,
+		i.config.Port,
+		params.SchedulingID,
+		params.CorrelationID)
+	payload := params.newRemoteReconciliationModel(callbackURL)
+
+	jsonPayload, err := json.Marshal(payload)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal HTTP payload to call reconciler of component '%s': %s", component, err)
+	}
+
+	compRecon, ok := i.config.Scheduler.Reconcilers[component]
+	if ok {
+		i.logger.Debugf("Remote invoker found dedicated reconciler for component '%s'", component)
+	} else {
+		i.logger.Debugf("Remote invoker found no dedicated reconciler for component '%s': "+
+			"using '%s' component reconciler as fallback", component, config.FallbackComponentReconciler)
+		compRecon, ok = i.config.Scheduler.Reconcilers[config.FallbackComponentReconciler]
+		if !ok {
+			i.logger.Errorf("Remote invoker could not find fallback reconciler '%s' in scheduler configuration",
+				config.FallbackComponentReconciler)
+			return nil, &NoFallbackReconcilerDefinedError{}
+		}
+	}
+
+	i.logger.Debugf("Remote invoker is calling remote reconciler via HTTP (URL: %s) "+
+		"for component '%s' (schedulingID:%s/correlationID:%s)",
+		compRecon.URL, params.ComponentToReconcile.Component, params.SchedulingID, params.CorrelationID)
+
+	resp, err := http.Post(compRecon.URL, "application/json", bytes.NewBuffer(jsonPayload))
+	if err == nil {
+		respDump, err := httputil.DumpResponse(resp, true)
+		if err == nil {
+			i.logger.Debugf("Remote invoker received HTTP response from reconciler of component '%s' with status '%s' [%d] "+
+				"(schedulingID:%s/correlationID:%s): %s",
+				params.ComponentToReconcile.Component, resp.Status, resp.StatusCode,
+				params.SchedulingID, params.CorrelationID, string(respDump))
+		} else {
+			i.logger.Warnf("Remote invoker failed to dump HTTP response from component reconciler: %s", err)
+		}
+	} else {
+		return resp, errors.Wrap(err, fmt.Sprintf("failed to call remote reconciler (URL: %s)", compRecon.URL))
+	}
+
+	return resp, nil
+}
+
+func (i *RemoteReconcilerInvoker) marshalHTTPResponse(body []byte, respModel interface{}, params *Params) error {
```

Comment:
Should be `unmarshalHTTPResponse` instead, right?

Reply:
Right!  :)

Reply:
fixed